From: Andreas Stieger <Andreas.Stieger@gmx.de>
Date: Wed, 03 May 2023 01:50:32 +0200
Subject: [PATCH] disable privacy-defeaing reporting feature
Upstream: no

The upstream developer has seen fit to have this program by default,
during first start-up and every subsequent run, without warning or
transport encryption, to report the following data to his server:

* a unique and persistent install ID
* the installed version
* the OS and python version in use

Additionaly it reported the following items during interactive use:

* All of the above data
* the radio make and model
* the operation performed on the radio
* (implicitly) the time and sequence of these operations

This patch disables this functionality.

Index: chirp-20240807/chirp/wxui/main.py
===================================================================
--- chirp-20240807.orig/chirp/wxui/main.py
+++ chirp-20240807/chirp/wxui/main.py
@@ -925,6 +925,7 @@ class ChirpMain(wx.Frame):
         self.Bind(wx.EVT_MENU, self._menu_about, about_item)
         help_menu.Append(about_item)
 
+        '''
         developer_menu = wx.MenuItem(help_menu, wx.NewId(),
                                      _('Developer Mode'),
                                      kind=wx.ITEM_CHECK)
@@ -966,6 +967,7 @@ class ChirpMain(wx.Frame):
                                 _('Load module from issue...'))
         self.Bind(wx.EVT_MENU, self._menu_load_from_issue, lmfi_menu)
         help_menu.Append(lmfi_menu)
+        '''
 
         self.bug_report_item = wx.MenuItem(
             help_menu, wx.NewId(),
Index: chirp-20240807/chirp/wxui/report.py
===================================================================
--- chirp-20240807.orig/chirp/wxui/report.py
+++ chirp-20240807/chirp/wxui/report.py
@@ -32,7 +32,7 @@ CONF = config.get()
 LOG = logging.getLogger(__name__)
 logging.getLogger('urllib3.connectionpool').setLevel(logging.INFO)
 SESSION = None
-DISABLED = False
+DISABLED = True
 SEM = threading.Semaphore(2)
 BASE = 'http://chirpmyradio.com/report'
 
